### PR TITLE
add dependabot support for npm and go dependencies

### PR DESCRIPTION
**Pull request by uncledru**
## Description

Adds dependabot support to `corso/src` (go). 
Adds dependabot support to `corso/website` (npm). 
Adds dependabot support to `corso/docs` (npm). 

Adds dependabot support to `corso` for actions as well.  

Warning that dependabot emails can get noisy so everyone may need to add email rules/update repository notifications 😭 

~PR limit of `5` here to test and we can increase as needed.~ Working well in `ark`. Increased to 50 w/ auto-merge. 

## Type of change

Please check the type of change your PR introduces:
- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :hamster: Trivial/Minor
- [x] :moneybag: Tech Debt

## Linear Issue(s)

- Fixes ALC-541

## Test Plan
Merge and see what happens (no way to currently test dependabot config). 

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E
- [x] :x: TEST IN PROD 👀

### CODE REVIEW

**Review by ryanfkeepers — APPROVED** [review]

**Review by ashmrtn — COMMENTED** [review]
Quick question: we want to have full control over the upgrade of some of our go dependencies (the ms graph code and Kopia come to mind). Sometimes upgrading these could cause things to break in ways that can only be discovered with more thorough testing outside of CI. Will this interfere with that?

**Comment by uncledru** [comment]
> Quick question: we want to have full control over the upgrade of some of our go dependencies (the ms graph code and Kopia come to mind). Sometimes upgrading these could cause things to break in ways that can only be discovered with more thorough testing outside of CI. Will this interfere with that?

Great question! Yes, it would update those dependencies. 

If we agree that critical dependencies should only be updated manually - I can tell dependabot to [ignore those deps](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#ignore).

**Comment by ashmrtn** [comment]
>If we agree that critical dependencies should only be updated manually - I can tell dependabot to [ignore those deps](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#ignore).

@vkamra @dadams39 should we add kopia and all the ms graph stuff to the ignore list?

**Comment by vkamra** [comment]
> > If we agree that critical dependencies should only be updated manually - I can tell dependabot to [ignore those deps](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#ignore).
> 
> @vkamra @dadams39 should we add kopia and all the ms graph stuff to the ignore list?

Yep - let's just remove automerge on `go` dependencies for now please.

File: .github/actions/auto-merge.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: auto-merge
+
+on:
+  pull_request:
+    paths-ignore:
+      - "src/**" # prevent auto-merge for go dependencies
```
Comment (uncledru) [inline .github/actions/auto-merge.yml:6]: Unfortunately auto-merge [config](https://github.com/marketplace/actions/dependabot-auto-merge#configuration-file-syntax) does not support [`ignore` options](https://github.com/ahmadnassri/action-dependabot-auto-merge/pull/43). 

[paths-ignore](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-paths) prevents the auto-merge action from running for `go` dependencies under `src/`. 

image

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+version: 2
+updates:
+  # Maintain dependencies for GitHub Actions
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+    reviewers:
+      - "alcion/platform"
```
Comment (vkamra) [inline .github/dependabot.yml:11]: Should this also be Corso maintainers?

**Review by vkamra — APPROVED** [review]
LGTM - there's one comment to look at please

**Pull request merged**